coderd/database/dbmem/dbmem.go

```diff
@@ -3758,6 +3758,32 @@ func (q *FakeQuerier) GetUserLinksByUserID(_ context.Context, userID uuid.UUID)
 	return uls, nil
 }
 
+func (q *FakeQuerier) GetUserWorkspaceBuildParameters(_ context.Context, params database.GetUserWorkspaceBuildParametersParams) ([]database.GetUserWorkspaceBuildParametersRow, error) {
+	q.mutex.RLock()
+	defer q.mutex.RUnlock()
+
+	userWorkspaceIDs := make(map[uuid.UUID]struct{})
+	for _, ws := range q.workspaces {
+		if ws.OwnerID != params.OwnerID {
+			continue
+		}
+		if ws.TemplateID != params.TemplateID {
+			continue
+		}
+		userWorkspaceIDs[ws.ID] = struct{}{}
+	}
+
+	userWorkspaceBuildParameters := make([]database.GetUserWorkspaceBuildParametersRow, 0)
+	for _, wbp := range q.workspaceBuildParameters {
+		userWorkspaceBuildParameters = append(userWorkspaceBuildParameters, database.GetUserWorkspaceBuildParametersRow{
```

Comment:
this returns every build parameter for every user, and also includes a named parameter multiple times if there are multiple builds.  You need to match the join and filtering behavior of the SQL query

Reply:
Built out the unit test a little bit in the interest of getting this wrapped up today.